git_hooks/hooks/pre-commit

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
```

Comment:
and I am unsure about the file name... shouldnt it be something like `.git/hooks/pre-commit` ?!

Reply:
.git directory cannot be pushed that why we have new_repo which has our hooks and then we tell users to use `git init --template 'git_hooks'` to places these hooks in .git/hooks
@goern

Reply:
erm... right... :/

Reply:
Anyway, what about keeping this hidden (`.git_hooks/pre-commit` - IMHO no need to have it nested - judging based on its name).

We do it for all configuration files and these hooks are something that will not be edited (often) by devs.

Reply:
@fridex  the nesting is needed by the ```git init command```. if the pre-commit is only inside git_hooks, then after reinitiating the git. The scripts will not move to .git/hooks but will only fo inside .git  which is of no use for us.

We can change the folder/dir name to ```.git_hooks``` from ```git_hooks```. 
Initially had kept it like `.git_hooks`,then changed it.